R/module_teal_data.R

```diff
@@ -100,109 +156,123 @@ srv_validate_reactive_teal_data <- function(id, # nolint: object_length
+srv_validate_error <- function(id, data, validate_shiny_silent_error) {
+  checkmate::assert_string(id)
+  checkmate::assert_flag(validate_shiny_silent_error)
+  moduleServer(id, function(input, output, session) {
+    output$message <- renderUI({
+      is_shiny_silent_error <- inherits(data(), "shiny.silent.error") && identical(data()$message, "")
+      if (inherits(data(), "error")) {
+        if (is_shiny_silent_error && !validate_shiny_silent_error) {
           return(NULL)
+        }
+        msg <- if (is_shiny_silent_error) {
+          "Check your inputs or contact app developer if error persists."
         } else {
-          validate(
-            need(
-              FALSE,
-              paste(
-                "Shiny error when executing the `data` module",
-                "Check your inputs or contact app developer if error persists.",
-                collapse = "\n"
-              )
-            )
-          )
+          data()$message
         }
+        validate(
+          need(
+            FALSE,
+            sprintf("Shiny error when executing the `data` module\n%s", msg)
+          )
+        )
       }
+    })
+  })
+}
 
-      # to handle errors and qenv.error(s)
-      if (inherits(data_out, c("qenv.error", "error"))) {
+#' @keywords internal
+ui_validate_qenv_error <- function(id) {
+  ns <- NS(id)
+  uiOutput(ns("message"))
+}
+
+#' @keywords internal
+srv_validate_qenv_error <- function(id, data) {
+  checkmate::assert_string(id)
+  moduleServer(id, function(input, output, session) {
+    output$message <- renderUI({
+      if (inherits(data(), c("qenv.error"))) {
         validate(
           need(
             FALSE,
-            paste0(
+            paste(
               "Error when executing the `data` module:",
-              strip_style(paste(data_out$message, collapse = "\n")),
-              "Check your inputs or contact app developer if error persists.",
+              strip_style(paste(data()$message, collapse = "\n")),
+              "\nCheck your inputs or contact app developer if error persists.",
               collapse = "\n"
             )
           )
         )
       }
+    })
+  })
+}
 
+#' @keywords internal
+ui_check_class_teal_data <- function(id) {
+  ns <- NS(id)
+  uiOutput(ns("message"))
+}
+
+#' @keywords internal
+srv_check_class_teal_data <- function(id, data) {
+  checkmate::assert_string(id)
+  moduleServer(id, function(input, output, session) {
+    output$message <- renderUI({
       validate(
         need(
-          checkmate::test_class(data_out, "teal_data"),
-          paste0(
-            "Assertion on return value from the 'data' module failed:",
-            checkmate::test_class(data_out, "teal_data"),
-            "Check your inputs or contact app developer if error persists.",
-            collapse = "\n"
-          )
+          inherits(data(), c("teal_data", "error")),
+          "Did not recieve a valid `teal_data` object. Cannot proceed further."
```

Comment:
Let's remove `a valid`. It's ambiguous. We had `valid` in here where this check also checked `datanames` and `emptiness`. Now it checks only class. Also there was a typo in `receive`.
```suggestion
          "Did not receive `teal_data` object. Cannot proceed further."
```